File: problems/2348-number-of-zero-filled-subarrays.ts
function zeroFilledSubarray(nums: number[]): number {
  let count = 0
  let index = 0
  while (index < nums.length) {
    if (nums[index] === 0) {
      const zeroStart = index
      let zeroEnd = nums.length
      for (let j = zeroStart + 1; j < nums.length; j++) {
        if (nums[j] !== 0) {
          zeroEnd = j
          break
        }
      }
      const n = zeroEnd - zeroStart
      count += (n * (n + 1)) / 2
      index = zeroEnd + 1
    } else {
      index++
    }
  }
  return count
}

File: problems/2348_number_of_zero_filled_subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: list[int]) -> int:
        count = 0
        cur_len = 0
        for num in nums:
            if num == 0:
                cur_len += 1
                count += cur_len
            else:
                cur_len = 0
        return count
